graphql/resolver.go

```diff
@@ -325,5 +325,22 @@ func New(apiURL string) Config {
 	c.Directives.RedactSecrets = func(ctx context.Context, obj any, next graphql.Resolver) (res any, err error) {
 		return next(ctx)
 	}
+	c.Directives.RequireAdmin = func(ctx context.Context, obj any, next graphql.Resolver) (res any, err error) {
+		dbUser := mustHaveUser(ctx)
+
+		permissions := gimlet.PermissionOpts{
+			Resource:      evergreen.SuperUserPermissionsID,
+			ResourceType:  evergreen.SuperUserResourceType,
+			Permission:    evergreen.PermissionAdminSettings,
+			RequiredLevel: evergreen.AdminSettingsEdit.Value,
+		}
+
+		if dbUser.HasPermission(permissions) {
+			return next(ctx)
+		}
+
+		return nil, InternalServerError.Send(ctx, "user lacks required permissions")
```

Comment:
let's also include the username in the message to see who is trying to access admin settings